src/common-utils/main.sh

```diff
@@ -450,57 +461,60 @@ if [ "${INSTALL_ZSH}" = "true" ]; then
                 -c receive.fsck.zeroPaddedFilemode=ignore \
-                "https://github.com/ohmyzsh/ohmyzsh" "${oh_my_install_dir}" 2>&1
-
+                "https://github.com/ohmyzsh/ohmyzsh" "${user_omz_install_dir}" 2>&1
             # Shrink git while still enabling updates
-            cd "${oh_my_install_dir}"
-            git repack -a -d -f --depth=1 --window=1
+            GIT_WORK_TREE="${user_omz_install_dir}" GIT_DIR="${user_omz_install_dir}/.git" git repack\
+                -a -d -f --depth=1 --window=1
         fi
 
-        # Add Dev Containers theme
-        mkdir -p ${oh_my_install_dir}/custom/themes
-        cp -f "${FEATURE_DIR}/scripts/devcontainers.zsh-theme" "${oh_my_install_dir}/custom/themes/devcontainers.zsh-theme"
-        ln -sf "${oh_my_install_dir}/custom/themes/devcontainers.zsh-theme" "${oh_my_install_dir}/custom/themes/codespaces.zsh-theme"
+        # Add dev containers theme
+        user_omz_theme_filepath="${user_omz_install_dir}/custom/themes"
+        user_devcontainer_theme_target="${user_omz_theme_filepath}/devcontainers.zsh-theme"
+        user_codespaces_theme_target="${user_omz_theme_filepath}/codespaces.zsh-theme"
+        theme_template_path="${FEATURE_DIR}/scripts/devcontainers.zsh-theme"
+        mkdir -p "${user_omz_theme_filepath}"
+        cp -f "${theme_template_path}" "${user_devcontainer_theme_target}"
+        cp -f "${theme_template_path}" "${user_codespaces_theme_target}"
 
-        # Add devcontainer .zshrc template
         if [ "$INSTALL_OH_MY_ZSH_CONFIG" = "true" ]; then
-            echo -e "$(cat "${template_path}")\nDISABLE_AUTO_UPDATE=true\nDISABLE_UPDATE_PROMPT=true" > ${user_rc_file}
+            # Add devcontainer .zshrc template
+            omz_rc_filename=".zshrc"
+            user_rc_file="${user_home}/${omz_rc_filename}"
+            omz_zshrc_template_path="${user_omz_install_dir}/templates/zshrc.zsh-template"
+            echo -e "$(cat "${omz_zshrc_template_path}")\nDISABLE_AUTO_UPDATE=true\nDISABLE_UPDATE_PROMPT=true" > ${user_rc_file}
             sed -i -e 's/ZSH_THEME=.*/ZSH_THEME="devcontainers"/g' ${user_rc_file}
+            omz_added_filesnames+=("${omz_rc_filename}")
         fi
+        # TODO remove installed files from previous step
+
+        user_omz_filepaths=( "${omz_added_filesnames[@]/#/$user_home/}" )
 
-        # Copy to non-root user if one is specified
         if [ "${USERNAME}" != "root" ]; then
-            copy_to_user_files=("${oh_my_install_dir}")
-            [ -f "$user_rc_file" ] && copy_to_user_files+=("$user_rc_file")
-            cp -rf "${copy_to_user_files[@]}" /root
-            chown -R ${USERNAME}:${group_name} "${oh_my_install_dir}" "${user_rc_file}"
+            # Copy files to alternate user if one is specified
+            cp -rf "${user_omz_filepaths[@]}" /root
+            # Set permissions for root user
+            chown -R root:root "${omz_added_filesnames[@]/#//root/}"
```

Comment:
Curious, why is this needed?

Reply:
Security. User scripts making it into root and being sourced.

Ie custom shell scripts
Or scripts appended to users zshrc

Nvm. I will close pr.